thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/AnomalyResource.java

```diff
@@ -440,6 +444,50 @@ public Response updateAnomalyFunction(@NotNull @QueryParam("id") Long id,
     return Response.ok(id).build();
   }
 
+  @POST
+  @Path("/anomaly-function/apply/{autotune_config_id}")
+  public Response applyReplayConfig(@PathParam("autotune_config_id") @NotNull long id,
+      @QueryParam("cloneFunction") @DefaultValue("false") boolean isCloneFunction,
+      @QueryParam("cloneAnomalies") @DefaultValue("false") boolean isCloneAnomalies) {
+    AutotuneConfigDTO autotuneConfigDTO = autotuneConfigDAO.findById(id);
+    if(autotuneConfigDTO == null) {
+      return Response.status(Response.Status.BAD_REQUEST).build();
+    }
+    if(autotuneConfigDTO.getConfiguration() == null) {
+      return Response.ok().build();
+    }
+
+    AnomalyFunctionDTO originalFunction = anomalyFunctionDAO.findById(autotuneConfigDTO.getFunctionId());
+    AnomalyFunctionDTO targetFunction = originalFunction;
+
+    // clone anomaly function and its anomaly results if requested
+    if(isCloneFunction) {
+      OnboardResource onboardResource = new OnboardResource(anomalyFunctionDAO, mergedAnomalyResultDAO, rawAnomalyResultDAO);
+      long cloneId;
+      String tag = "clone";
+      try {
+        cloneId = onboardResource.cloneFunctionsGetIds(originalFunction.getId(), "clone", Boolean.toString(isCloneAnomalies));
+      } catch (Exception e) {
+        LOG.warn("Unable to clone function {} with clone tag \"{}\"", originalFunction.getId());
+        return Response.status(Response.Status.CONFLICT).build();
+      }
+      targetFunction = anomalyFunctionDAO.findById(cloneId);
+    }
+
+    // Update function configuration
+    targetFunction.updateProperties(autotuneConfigDTO.getConfiguration());
+    targetFunction.setActive(true);
+    anomalyFunctionDAO.update(targetFunction);
+
+    // Dis-activate original function
+    if(isCloneFunction) {
+      originalFunction.setActive(false);
```

Comment:
How do we deal with the anomalies from original function? It varies with the usecase:
1. The cloned function is for replay and replace original function. You don't care about the anomalies from original function. You can erase the historical anomalies from original function
2. If original function already have labeled anomalies, then you should always have "isCloneAnomalies = true"